iothub/device/src/RetryPolicies/NoRetry.cs

```diff
@@ -10,6 +11,16 @@ namespace Microsoft.Azure.Devices.Client
     /// </summary>
     public class NoRetry : IRetryPolicy
     {
+        /// <summary>
+        /// Create an instance of a retry policy that perfrms no retries.
+        /// </summary>
+        public NoRetry()
+        {
+            if (Logging.IsEnabled)
+                Logging.Info(this, $"NOTE: A no-retry retry policy has been enabled," +
+                    $" the device client will not perform any retries on disconnection.", nameof(NoRetry));
```

Comment:
This is the only line of code being added

Reply:
nit: wouldn't a module client also hit this code path? Maybe just "... the ~~device~~ client will not..."

Reply:
That's true, I'll update it!